source/fundamentals/aggregation/aggregation-expression-operations.txt

```diff
@@ -282,6 +282,108 @@ the Query API:
+       | `passBooleanTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlBoolean.html#passBooleanTo(java.util.function.Function)>`__
+       | `passDateTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDate.html#passDateTo(java.util.function.Function)>`__
+       | `passDocumentTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlDocument.html#passDocumentTo(java.util.function.Function)>`__
+       | `passIntegerTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlInteger.html#passIntegerTo(java.util.function.Function)>`__
+       | `passMapTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlMap.html#passMapTo(java.util.function.Function)>`__
+       | `passNumberTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlNumber.html#passNumberTo(java.util.function.Function)>`__
+       | `passStringTo() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/mql/MqlString.html#passStringTo(java.util.function.Function)>`__
+
+     - :manual:`$function </reference/operator/aggregation/function/>`
+
+.. include:: /includes/fundamentals/agg-mql-diff.rst
+
+Suppose you need to determine how a class is performing against some
+benchmarks. You want to find the average final grade for each class and
+compare it against the benchmark values to identify whether the class
+does not meet, meets, or exceeds expectations.
+
+To calculate each class average, write an averaging method called
+``classAvg()`` and pass the ``students`` array to this method by using
+the ``passArrayTo()`` method. To compare each class average against your
+defined benchmarks, write a method called ``evalAvg()`` and apply this
+method to the class averages by chaining the ``passNumberTo()`` method.
+To output the result as the ``evaluation`` field, use the ``project()``
+method.
+
+The following code shows the pipeline stage for this aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   Arrays.asList(project(fields(
+           computed("evaluation",
+                   current().<MqlDocument>getArray("students")
+                           .passArrayTo(u -> classAvg(u))
+                           .passNumberTo(v -> evalAvg(v))
+   ))));
+
+The following code shows sample implementations of the methods that were
+used in the previous aggregation:
+
+.. code-block:: java
+   :copyable: true
+
+   public static MqlNumber classAvg(MqlArray<MqlDocument> u) {
+       var sum = u.sum(x -> x.getInteger("finalGrade"));
+       var avg = sum.divide(u.size());
+       return avg;
+   }
+   
+   public static MqlString evalAvg(MqlNumber v) {
+       MqlNumber[] b = {of(70), of(85)};
+       var msg = v.switchOn(on -> on
+               .lte(b[0], m -> of("Needs improvement"))
+               .lte(b[1], m -> of("Meets expectations"))
+               .defaults(m -> of("Exceeds expectations")));
+       return msg;
+   }
```

Comment:
[question]

Would it make sense to put these methods before the pipeline? That way readers already know what they are when they are pulled into the pipeline?